express/src/decorators/controller.ts

```diff
@@ -7,14 +7,15 @@ import { Type } from '../middleware';
  * Registers controller for base url
  *
  * @param {string} url
+ * @param routerOptions
  * @param {Type[]} [middleware]
  */
 export function Controller(url: string, routerOptions?: Type[] | RouterOptions, middleware?: Type[]) {
   return (target): void => {
     const meta: ExpressMeta = getMeta(target.prototype);
 
     meta.url = url;
-    meta.middleware = Array.isArray(routerOptions) ? routerOptions : middleware;
+    meta.middleware = Array.isArray(routerOptions) ? routerOptions.concat(meta.middleware) : middleware.concat(meta.middleware);
```

Comment:
Could you add just in case safe check, to be sure, that middleware exists, i.e. `.concat(meta.middleware || [])`...

Reply:
Done. I also added overload signatures for the two variants